aggregator/src/aggregator/aggregation_job_driver.rs

```diff
@@ -641,18 +645,34 @@ where
                 .get(RETRY_AFTER)
                 .map(parse_retry_after)
                 .transpose()?;
-            let resp = AggregationJobResp::get_decoded(http_response.body())
-                .map_err(Error::MessageDecode)?;
+            let length = http_response
+                .headers()
+                .get(CONTENT_LENGTH)
+                .map(parse_content_length)
+                .transpose()
+                .map_err(|e| Error::BadRequest(e.into()))?;
+
+            let resp = if length.is_some_and(|l| l > 0) {
```

Comment:
This won't work if we are talking to a helper that streams its responses, and doesn't set `Content-Length`. We could instead call `http_response.body()` sooner, and check its length.

Reply:
I've changed this to check the body.